src/pages/index.js

```diff
@@ -25,39 +28,55 @@ const cardSection = new Section(
 
 const profileEditButton = document.querySelector("#profile-edit-button");
 
-const handlePopupWithForm = new PopupWithForm("#profile-edit-modal", (data) => {
-  userInfo.setUserInfo({
-    name: data.title,
-    job: data.description,
-  });
-  handlePopupWithForm.close();
+const editProfilePopup = new PopupWithForm("#profile-edit-modal", (data) => {
+  editProfilePopup.renderModalFormLoading(true);
+  api
+    .editProfile(data.title, data.description)
+    .then((updatedUserInfo) => {
+      userInfo.setUserInfo({
+        name: updatedUserInfo.name,
+        job: updatedUserInfo.about,
+      });
+      editProfilePopup.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => editProfilePopup.renderModalFormLoading(false));
 });
 
-handlePopupWithForm.setEventListeners();
+profileEditButton.addEventListener("click", () => {
+  const currentUserData = userInfo.getUserInfo();
+  editProfilePopup.setInputValues({ currentUserData });
+  editProfilePopup.open();
+});
+
+editProfilePopup.setEventListeners();
 
 profileEditButton.addEventListener("click", () => {
   const currentUserData = userInfo.getUserInfo();
-  handlePopupWithForm.setInputValues({
+  editProfilePopup.setInputValues({
     title: currentUserData.name,
     description: currentUserData.job,
   });
-  handlePopupWithForm.open();
+  editProfilePopup.open();
 });
 
 const addCardForm = document.forms["Add-a-New-Card"];
 addCardForm.reset();
 
 const addCardWithPopupForm = new PopupWithForm("#add-card-modal", (data) => {
-  cardSection.addItem(
-    createCard({
-      name: data.title,
-      link: data.url,
+  addCardWithPopupForm.renderModalFormLoading(true);
+  api
+    .createNewCard(data.title, data.url)
+    .then((newCard) => {
+      cardSection.addItem(createCard(newCard));
+      addCardWithPopupForm.close();
+      addCardForm.reset();
+      formValidators["Add-a-New-Card"].disableButton();
     })
-  );
-  addCardWithPopupForm.close();
-  addCardForm.reset();
-  formValidators["Add-a-New-Card"].disableButton();
+    .catch((err) => console.error(err))
+    .finally(() => addCardWithPopupForm.renderModalFormLoading(false));
 });
+
 addCardWithPopupForm.setEventListeners();
 
 const handlePopupWithImage = new PopupWithImage("#image-preview-modal");
```

Comment:
`handlePopupWithImage`


This is not a function, so it should not start with `handle`


`imageModal` or `popupWithImage`